### ISSUE TITLE
First  doesn't work after sort

### ISSUE THREAD

**Issue by tkozybski** [simonask/rx-ranges#8]
As in the title, code below doesn't compile without calling to_vector() after sort

```
    std::vector<int> vec { 1, 2, 3, 4, 5 };

    auto firstItem = vec
        | rx::sort()
        | rx::first();
```

**Comment by tkozybski** [comment]
I know min_element could be use in this scenario but my use case was like this to get optional element in case everything's filtered out:  vec | filter(condition) | sort() | first

**Comment by Hazurl** [comment]
Fixed in #10
